File: ifgi/scene/test_Film.py
#!/usr/bin/env python
#
#
# Copyright (C) 2010-2011 Yamauchi, Hitoshi
#
# test for Film module
#

"""test IFGI Film module"""

import unittest
import numpy
import Film

# test Film
class TestFilm(unittest.TestCase):
    """test suit for Film."""
    def test_imagefilm(self):
        """test for ImageFilm"""
        f = Film.ImageFilm((128, 128, 4), 'RGBA')
        assert(str(f) == '[name: RGBA, resolution: (128 128 4)]')

        res = f.get_resolution()
        assert(res == (128, 128, 4))

        # fill white
        f.fill_color(numpy.array([1,1,1,1]))

        # draw a line
        red = numpy.array([1, 0, 0, 1])
        for i in xrange(10, 100, 1):
            f.put_color((i, i), red)
            assert((f.get_color((i, i)) == red).all())

        # save a file
        f.save_file('test_film_result.png')

#
# main test
#
if __name__ == '__main__':
    suit0   = unittest.TestLoader().loadTestsFromTestCase(TestFilm)
    alltest = unittest.TestSuite([suit0])
    unittest.TextTestRunner(verbosity=2).run(alltest)

File: ifgi/cpp/scene/test_Film.cpp
//----------------------------------------------------------------------
// ifgi c++ implementation: test_Film.py
// Copyright (C) 2010-2012 Yamauchi, Hitoshi
//----------------------------------------------------------------------
/// \file
/// \brief test_Film.py

#include "ImageFilm.hh"

#include <string>
#include <gtest/gtest.h>


/// Test ImageFilm
/// TEST(test_case_name, test_name)
TEST(ImageFilmTest, ImageFilmLineSaveTest)
{
    ifgi::ImageFilm film(ifgi::Sint32_3(128, 100, 4), "RGBA");
    ASSERT_EQ(film.to_string(), std::string("[name: RGBA, resolution: (128 100 4)]"));

    ifgi::Sint32_3 const res = film.get_resolution();
    ASSERT_EQ(res, ifgi::Sint32_3(128, 100, 4));

    // fill white color
    film.fill_color(ifgi::Color(1.0, 1.0, 1.0, 1.0));

    // draw a line
    ifgi::Color const red(1.0, 0.0, 0.0, 1.0);
    for(ifgi::Sint32 i = 10; i < 90; ++i){
        film.put_color(i, i, red);
        ASSERT_EQ(film.get_color(i, i), red);
    }

    // save a file
    ASSERT_EQ(film.save_file("test_film_result.ppm", "ppm"), true);
    ASSERT_EQ(film.save_file("test_film_result.gfi", "gfi"), true);
}

int main(int argc, char **argv)
{
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}
